test/index.test.ts

```diff
@@ -618,6 +618,262 @@ describe('HipThrusTS', () => {
 
         const triple = HTPipe(left, midNotCovered, rightFullyCovered);
       });
```

Comment:
what we really want to test here is that each sanitize that is added to the pipe ONLY SUBTRACTS - i.e. doesn't allow stuff from previous stages to get through

so leave left alone, but change right such that:

- [ ] make sure right's incoming context doesn't have a someObj property
- [ ] have right return ONLY context.b
- [ ] make sure the FINAL piped sanitizedParams ONLY has b

Reply:
Also, unless I'm missing something, some of these tests seem to be overlapping the other tests (which we don't have for these lifecycles I know) - keep these tests independent and focused only for checking the "filtration" functionality.

...and do the above bullets for all the tests

I guess a new PR would be better.

As for another PR for doing the "normal" tests for all lifecycle stages, let's do some thinking for a separate PR.  Some factoring is probably possible.  If test code gets too big and repetitive it can create bugs.

Reply:
about the other tests - I added this https://github.com/captaincaius/hipthrusts/issues/21#issuecomment-631655727

Reply:
in fact... maybe you can make this first "SanitizerSubtractsContextInsteadOfAdds" your first guinea pig for that approach - up to you.

Reply:
done in this PR: https://github.com/captaincaius/hipthrusts/pull/78/files